tests/integration_tests/functional/test_net.py

```diff
@@ -78,3 +78,65 @@ def test_multi_queue_unsupported(uvm_plain):
             host_dev_name=tapname,
             guest_mac="AA:FC:00:00:00:01",
         )
+
+
+def run_udp_offload_test(vm):
+    """
+    - Start a socat UDP server in the guest.
+    - Try to send a UDP message with UDP offload enabled.
+
+    If tap offload features are not configured, an attempt to send a message will fail with EIO "Input/output error".
+    More info (search for "TUN_F_CSUM is a must"): https://blog.cloudflare.com/fr-fr/virtual-networking-101-understanding-tap/
+    """
+    port = "81"
+    out_filename = "/tmp/out.txt"
+    message = "x"
+
+    # Start a UDP server in the guest
+    # vm.ssh.check_output(f"nohup socat UDP-LISTEN:{port} - > {out_filename} &")
+    vm.ssh.check_output(
+        f"nohup socat UDP-LISTEN:{port} OPEN:{out_filename},creat > /dev/null 2>&1 &"
+    )
+
+    # Try to send a UDP message from host with UDP offload enabled
+    cmd = f"ip netns exec {vm.ssh_iface().netns} python3 ./host_tools/udp_offload.py {vm.ssh_iface().host} {port}"
```

Comment:
```
    cmd = f"{vm.netns.cmd_prefix} ... "
```


Also I guess the only reason it's a separate script is because we need to change network namespace?

Reply:
yes, exactly